### PR TITLE
feature: enable typescript isolated modules

### LINKED ISSUE #212835: feature request: Enabling typescript isolated modules
Related #160416

For migrating towards ESM in VSCode, it seems the tsconfig [isolatedModules](https://www.typescriptlang.org/tsconfig/#isolatedModules) option could be useful, ensuring that imports are compatible with ESM. 

For example, this typescript code transpiled to javascript would cause a runtime error because of the `someType` export:

```js
import { someType, someFunction } from "someModule";
 
someFunction();

export { someType, someFunction };
 ```

The `isolatedModules` rule would enforce a type only import for `someType`:

```js
import { type someType, someFunction } from "someModule";
 
someFunction();

export { type someType, someFunction };
```

Enabling the `isolatedModules` tsconfig option in VSCode, it seems these files would need to adjusted:

```
Found 26 errors in 6 files.

Errors  Files
     9  src/vs/base/common/amd.ts:57
     4  src/vs/base/node/processes.ts:14
     1  src/vs/workbench/contrib/accessibility/browser/accessibilityConfiguration.ts:636
     1  src/vs/workbench/contrib/inlineChat/common/inlineChat.ts:172
     8  src/vs/workbench/contrib/performance/browser/perfviewEditor.ts:331
     3  src/vs/workbench/contrib/tasks/common/taskService.ts:18
```

### PR DESCRIPTION
**Pull request by SimonSiefke**
Fixes #212835

### CODE REVIEW

File: src/vs/workbench/contrib/inlineChat/common/inlineChat.ts [hunk 1]
```diff
@@ -148,6 +148,11 @@ export interface IInlineChatService {
 export const INLINE_CHAT_ID = 'interactiveEditor';
 export const INTERACTIVE_EDITOR_ACCESSIBILITY_HELP_ID = 'interactiveEditorAccessiblityHelp';
 
+export const enum EditMode {
```
Comment (jrieken) [inline src/vs/workbench/contrib/inlineChat/common/inlineChat.ts:151]: unrelated change?

**Review by jrieken — CHANGES_REQUESTED** [review]
Thanks. This is looking good and I am surprised that TSC doesn't complain about all our const-enum usages

File: src/vs/workbench/contrib/inlineChat/common/inlineChat.ts (hunk 1 above)
Comment (SimonSiefke) [reply src/vs/workbench/contrib/inlineChat/common/inlineChat.ts:151]: No, it is intentional. It fixes this typescript error: `Enum 'EditMode' used before its declaration`:

enum-used-before-declaration

**Comment by jrieken** [comment]
> https://www.typescriptlang.org/tsconfig/#references-to-const-enum-members

👀  reading this again and some stackoverflow help resolves the mystery: only ambient/declared const enums are an issue

**Review by jrieken — APPROVED** [review]

**Review by lszomoru — APPROVED** [review]
